LoginApp/LoginViewController.swift

```diff
@@ -32,45 +35,69 @@ class LoginViewController: UIViewController {
 		super.didReceiveMemoryWarning()
 	}
 	
-	@IBAction func checkLoginCredentials() {
+	func validate(email: String, password: String) -> (Bool,String) {
+		
+		if email.isEmpty || password.isEmpty {
+			
+			return (false,"Please provide data in all the fields.")
+			
+		}
+		if !isValid(email: email) {
+			
+			return (false,"Please provide valid email id.")
+			
+		}
+		return (true,"Success")
+	}
+	
+	func isValid(email:String) -> Bool {
 		
-//		let array = UserInfo.mr_findAll() as? [UserInfo]
-//		
-//		print(array?.first?.emailId)
+		let emailRegex = "[A-Z0-9a-z._%+-]+@[A-Za-z0-9.-]+\\.[A-Za-z]{2,6}"
+		
+		return NSPredicate(format: "SELF MATCHES %@", emailRegex).evaluate(with: email)
+		
+	}
 
-		if let name: String = userName.text, name.characters.count > 0, let tempPassword: String = password.text, tempPassword.characters.count > 0{
+	func showAlert(message: String) {
+		
+		let alertController = UIAlertController(title: "", message: message, preferredStyle: .alert)
+		
+		let OKAction = UIAlertAction(title: "OK", style: .default, handler: nil)
+		alertController.addAction(OKAction)
+		
+		self.present(alertController, animated: true, completion: nil)
+		
+	}
+	
+	@IBAction func checkLoginCredentials() {
+		
+		if let email = emailField.text, let password = passwordField.text {
 			
-			if let userDatabase:[String: String] = defaults.dictionary(forKey: "userDatabase") as? [String : String]  {
+			let response = validate(email: email, password: password)
+		
+			if !response.0 {
```

Comment:
use proper naming for tuples, here 0 and 1 represents nothing, until we dig further inside.